mopy/core/spectrumgroup.py

```diff
@@ -341,16 +492,65 @@ def fit_source_model(self, model="brune", fit_noise=False, **kwargs):
             The model, or sequence of models, to fit to the data
         fit_noise
             If True, also fit to the noise, else drop before fitting.
-
-        Returns
-        -------
-
         """
         fit = fit_model(self, model=model, fit_noise=fit_noise, **kwargs)
         return self.new_from_dict(fit_df=fit)
 
+    def calc_corner_frequency(
+        self, smoothing: Optional[str] = None, **kwargs
+    ) -> pd.Series:
+        """
+        Calculate the corner frequency of the displacement spectra
+
+        Parameters
+        ----------
+        smoothing
+            Smoothing to apply to the spectra
+
+        Other Parameters
+        ----------------
+        Other parameters applying to data smoothing may be provided
+
+        Note
+        ----
+        Calculation is actually performed by getting the maximum of the velocity spectra
+        """
+        sg = self.abs()
+        # Convert to velocity spectra
+        vel = sg.to_spectra_type(
+            "cft", motion_type="velocity", smoothing=smoothing, **kwargs
+        )
+        return vel.data.idxmax(axis=1)  # Corner frequency
+
+    def calc_omega0(self, fc: pd.Series, smoothing: Optional[str] = None) -> pd.Series:
+        """
+        Calculate Omega0 for the displacement spectra
+
+        Parameters
+        ----------
+        fc
+            Corner frequencies for each trace
+        smoothing
+            Name of the smoothing function to apply to the data, if any
+        """
+        sg = self.abs()
+        disp = sg.to_spectra_type(
+            "cft", motion_type="displacement", smoothing=smoothing
+        )
+        # TODO: Consider swapping this around to read logically (i.e., don't
```

Comment:
@d-chambers I have no idea why it was done like this; it could be something like that, though. I just know that when I was updating it this line wasn't working and it took me a good 2 hours to fully appreciate what this line of code was doing (even with the comment that explains exactly what the line of code is doing). 

If I remember right, it was partially because there's something weird about numpy.greater.outer that was making it a real beast to troubleshoot.